improver/cli/apply_dz_rescaling.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown copyright. The Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""CLI to apply a scaling factor to account for a correction linked to the
+difference in altitude between the grid point and the site location."""
+
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(
+    forecast: cli.inputcube,
+    scaling_factor: cli.inputcube,
+    *,
+    site_id_coord: str = "wmo_id",
+):
```

Comment:
```suggestion
) -> Cube:
```

Reply:
I don't think that we actually add the output type hint for the CLIs as I'd have to import `Cube` from iris and I think the aim is to import as little as possible outside the `process` method of the CLI. I think that the motivation for this is that someone, in principle, could look at the the help information for these CLIs on the command line using e.g. `bin/improver apply-dz-rescaling -h` and this will be quicker if the number of modules imported is as low as possible. [This test](https://github.com/metoppv/improver/blob/master/improver_tests/cli/test_init.py#L407) actually fails if I import Iris in a CLI.